examples/complicated_cpu_bound_benchmark.cf

```diff
@@ -0,0 +1,1031 @@
+      "parsejson"
+        usebundle => parsejson_bench("$(i)");
+      "parserealarray"
+        usebundle => parserealarray_bench("$(i)");
+      "parsestringarray"
+        usebundle => parsestringarray_bench("$(i)");
+      "parsestringarrayidx"
+        usebundle => parsestringarrayidx_bench("$(i)");
+      "product"
+        usebundle => product_bench("$(i)");
+      "random"
+        usebundle => randomint_bench("$(i)");
+      "regarray"
+        usebundle => regarray_bench("$(i)");
+      "regcmp"
+        usebundle => regcmp_bench("$(i)");
+      "regex_replace"
+        usebundle => regex_replace_bench("$(i)");
+      "regextract"
+        usebundle => regextract_bench("$(i)");
+      "reglist"
+        usebundle => reglist_bench("$(i)");
+      "reverse"
+        usebundle => reverse_bench("$(i)");
+      "shuffle"
+        usebundle => shuffle_bench("$(i)");
+      "some"
+        usebundle => some_bench("$(i)");
+      "sort"
+        usebundle => sort_bench("$(i)");
+      "splitstring"
+        usebundle => splitstring_bench("$(i)");
+      "storejson"
+        usebundle => storejson_bench("$(i)");
+      "strcmp"
+        usebundle => strcmp_bench("$(i)");
+      "strftime"
+        usebundle => strftime_bench("$(i)");
+      "string_downcase"
+        usebundle => string_downcase_bench("$(i)");
+      "string_head"
+        usebundle => string_head_bench("$(i)");
+      "string_length"
+        usebundle => string_length_bench("$(i)");
+      "string_mustache"
+        usebundle => string_mustache_bench("$(i)");
+      "string_replace"
+        usebundle => string_replace_bench("$(i)");
+      "string_reverse"
+        usebundle => string_reverse_bench("$(i)");
+      "string_split"
+        usebundle => string_split_bench("$(i)");
+      "string_tail"
+        usebundle => string_tail_bench("$(i)");
+      "string_upcase"
+        usebundle => string_upcase_bench("$(i)");
+      "sublist"
+        usebundle => sublist_bench("$(i)");
+      "sum"
+        usebundle => sum_bench("$(i)");
+      "translatepath"
+        usebundle => translatepath_bench("$(i)");
+      "unique"
+        usebundle => unique_bench("$(i)");
+      "validdata"
+        usebundle => validdata_bench("$(i)");
+      "validjson"
+        usebundle => validjson_bench("$(i)");
+      "variablesmatching"
+        usebundle => variablesmatching_bench("$(i)");
+      "variablesmatching_as_data"
+        usebundle => variablesmatching_as_data_bench("$(i)");
+      "variance"
+        usebundle => variance_bench("$(i)");
+}
+
+bundle agent accumulated_bench(i)
+{
+  vars:
+      "test"
+        int => accumulated("$(i)", "$(i)", "$(i)", "$(i)", "$(i)", "$(i)");
+}
+
+bundle agent ago_bench(i)
+{
+  vars:
+      "test"
+        int => ago("$(i)", "$(i)", "$(i)", "$(i)", "$(i)", "$(i)");
+}
+
+bundle agent and_bench(i)
+{
+  classes:
+      "test_1"
+        expression => and("any", "any");
+      "test_2"
+        expression => and("any", "!any");
```

Comment:
If you try to use `i`  as much as possible, you make can sure that function calls are not cached:

```suggestion
        expression => and("any", "any_$(i)");
```